codalab/model/bundle_model.py

```diff
@@ -221,9 +225,11 @@ def get_bundle_worker(self, uuid):
             row = conn.execute(
                 cl_worker_run.select().where(cl_worker_run.c.run_uuid == uuid)
             ).fetchone()
-            precondition(
```

Comment:
When you say the precondition error blocks the rest workers from getting acknowledged, how does that block actually work?

Reply:
I'm guessing it throws an exception, and the upstream caller can't handle that exception.

Reply:
It's from here [common.py](https://github.com/codalab/codalab-worksheets/blob/28e3c011f0cc426ea0dc27c7eeba844869b8159c/codalab/common.py#L93,) which raises an exception that will be eventually caught by [bundle_manager.py](https://github.com/codalab/codalab-worksheets/blob/master/codalab/server/bundle_manager.py#L66)

Reply:
Ah so is the catching of this exception taking a lot of time? What happens after, does the bundle that raise this exception continue to try getting acknowledgement and keep raising the exception, which therefore blocks the finalizing?

Reply:
It's once the exception is thrown, the rest of the code will be skipped and the main process will return back to this part [bundle_manager.py](https://github.com/codalab/codalab-worksheets/blob/master/codalab/server/bundle_manager.py#L66)

Reply:
I see